fix reproject raster layer using gdal

Gustry opened the pull request.

ismailsunni commented on safe/gisv4/raster/reproject.py:
```diff
@@ -36,18 +45,91 @@ def reproject(layer, output_crs, callback=None):
+
+
+def _reproject_gdal(layer, output_crs, callback=None):
+    """Reproject using GDAL command line.
```
Add param and type

ismailsunni commented on safe/gisv4/raster/reproject.py:
```diff
@@ -36,18 +45,91 @@ def reproject(layer, output_crs, callback=None):
+
+
+def _reproject_pyqgis(layer, output_crs, callback=None):
+    """Not working properly for now. Need more tests.
```
Add param and type.

ismailsunni commented on safe/gisv4/vector/reproject.py:
```diff
@@ -33,8 +35,8 @@ def reproject(layer, output_crs, callback=None):
     :type output_crs: QgsCoordinateReferenceSystem
```
Complete param and type.

Gustry replied:
Just expand the code, you will see it's there.

ismailsunni replied:
Ah, my bad

Gustry commented:
@ismailsunni to process a raster, we have 3 ways : 
- use the QGIS API only
- use the GDAL API
- use the GDAL command line

I tried to use to use the QGIS api, because it's better if we can use QgsRasterLayer everywhere. If I use GDAL API or GDAL command line, I need to work with a file on the file-system.
But it's not working properly. That's why I added this kind of wrapper to switch easily when I will find why it's not working with PyQGIS. The raster processing in pyqgis is not very well documented instead of GDAL.
So in the meantime, I added the GDAL command line to do the job.

Gustry commented:
We are not going to reproject raster. I'm going to close this PR and make a new one with only the code we need.
